src/main/java/org/mule/tools/apikit/model/CustomConfiguration.java

```diff
@@ -0,0 +1,54 @@
+/*
+ * Copyright (c) MuleSoft, Inc.  All rights reserved.  http://www.mulesoft.com
+ * The software in this package is published under the terms of the CPAL v1.0
+ * license, a copy of which has been included with this distribution in the
+ * LICENSE.txt file.
+ */
+package org.mule.tools.apikit.model;
+
+import java.util.List;
+import java.util.Optional;
+
+public class CustomConfiguration {
```

Comment:
I wouldn't create a different config class, this could be added at ScaffoldingConfiguration as optionals